File: test/python/test_1.py
import sys
import os

sys.path.append(os.getcwd())
from .python_deps.boilerplate import Test
from src.python.code_1 import Solution


def input_fetcher(case):
    return case["input"]["nums"], case["input"]["target"]


def test():
    test = Test(1, input_fetcher)
    solution = Solution().twoSum
    test.run_tests(solution)

File: test/cpp/test_1.cpp
#include "../../src/cpp/code_1.cpp"
#include "cpp_deps/boilerplate.hpp"

void test(Solution& sol, const json& input, const json& output) {
    vector<int> nums = input["nums"].get<vector<int>>();
    int target = input["target"];
    vector<int> expected = output.get<vector<int>>();
    vector<int> result = sol.twoSum(nums, target);
    sort(result.begin(), result.end());
    CHECK_EQ(result, expected);
}

TEST_CASE("") {
    TEST("test/test_json/test_1.json");
}